File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/CandidateTicket.java
package com.eulersbridge.isegoria.network.api.models;

import android.text.TextUtils;

import com.squareup.moshi.Json;

@SuppressWarnings("unused")
public class CandidateTicket {

    @Json(name = "ticketId")
    public long id;

    private String name;
    private String givenName;
    private String familyName;

    @Json(name = "numberOfSupporters")
    public String supportersCount;

    public String information;
    public String logo;

    private String colour;

    public String getName() {
        if (!TextUtils.isEmpty(name)) {
            return name;

        } else {
            return String.format("%s %s", givenName, familyName);
        }
    }

    public String getColour() {
        if (!TextUtils.isEmpty(colour)) {
            return colour;

        } else {
            return "#000000";
        }
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/CandidateTicket.kt
package com.eulersbridge.isegoria.network.api.models

import com.squareup.moshi.Json

data class CandidateTicket (

    @field:Json(name = "ticketId")
    var id: Long = 0,

    var name: String?,
    var givenName: String?,
    var familyName: String?,

    @field:Json(name = "numberOfSupporters")
    var supportersCount: String?,

    var information: String?,
    var logo: String? = null,

    private val colour: String?

) {
    val fullName: String
        get() = name?.takeUnless { it.isBlank() } ?: "$givenName $familyName"
    
    fun getColour()
        = colour?.takeUnless { it.isBlank() } ?: "#000000"
}